docs/topics/search/backends.md

```diff
@@ -0,0 +1,181 @@
+(wagtailsearch_backends)=
+
+# Backends
+
+Wagtailsearch has support for multiple backends, giving you the choice between using the database for search or an external service such as Elasticsearch.
+
+You can configure which backend to use with the `WAGTAILSEARCH_BACKENDS` setting:
+
+```python
+WAGTAILSEARCH_BACKENDS = {
+    'default': {
+        'BACKEND': 'wagtail.search.backends.database',
+    }
+}
+```
+
+(wagtailsearch_backends_auto_update)=
+
+## `AUTO_UPDATE`
+
+By default, Wagtail will automatically keep all indexes up to date. This could impact performance when editing content, especially if your index is hosted on an external service.
+
+The `AUTO_UPDATE` setting allows you to disable this on a per-index basis:
+
+```python
+WAGTAILSEARCH_BACKENDS = {
+    'default': {
+        'BACKEND': ...,
+        'AUTO_UPDATE': False,
+    }
+}
+```
+
+If you have disabled auto update, you must run the [](update_index) command on a regular basis to keep the index in sync with the database.
+
+(wagtailsearch_backends_atomic_rebuild)=
+
+## `ATOMIC_REBUILD`
+
+```{warning}
+This option may not work on Elasticsearch version 5.4.x, due to [a bug in the handling of aliases](https://github.com/elastic/elasticsearch/issues/24644) - please upgrade to 5.5 or later.
+```
+
+By default (when using the Elasticsearch backend), when the `update_index` command is run, Wagtail deletes the index and rebuilds it from scratch. This causes the search engine to not return results until the rebuild is complete and is also risky as you can't rollback if an error occurs.
+
+Setting the `ATOMIC_REBUILD` setting to `True` makes Wagtail rebuild into a separate index while keep the old index active until the new one is fully built. When the rebuild is finished, the indexes are swapped atomically and the old index is deleted.
+
+## `BACKEND`
+
+Here's a list of backends that Wagtail supports out of the box.
+
+(wagtailsearch_backends_database)=
+
+### Database Backend (default)
+
+`wagtail.search.backends.database`
+
+The database search backend searches content in the database using the full text search features of the database backend in use (such as PostgreSQL FTS, SQLite FTS5).
+This backend is intended to be used for development and also should be good enough to use in production on sites that don't require any Elasticsearch specific features.
+
+```{versionchanged} 2.15
+``wagtail.search.backends.database`` replaces the old ``wagtail.search.backends.db`` backend which works using simple substring matching only. ``wagtail.search.backends.db`` is still the default if ``WAGTAILSEARCH_BACKENDS`` is not specified; ``wagtail.search.backends.database`` will become the default in Wagtail 2.17.
```

Comment:
question should the inline code samples be in double backticks? or is this a side effect of MyST?

Reply:
nice pickup. Thanks. Fixed now.